Fabric.Authorization.AccessControl/src/app/services/global/services.service.ts

```diff
@@ -56,17 +67,32 @@ export class ServicesService {
                     const targetService: IDiscoveryService = response.value.find(
                         s => s.ServiceName === service.name && (!service.version || s.Version === service.version)
                     );
-                    if (targetService === undefined) {
+
+                    if (service.name === 'DiscoveryService') {
+                        service.url = this.discoveryServiceEndpoint;
+                    } else if (targetService === undefined) {
                         throw new Error(`The ${service.name} was not found in discovery service. Please ensure it is set up correctly.`);
+                    } else {
+                        service.url = targetService.ServiceUrl;
                     }
-                    service.url = targetService.ServiceUrl;
+                    
+                    return this.services.find(s => s.name === 'IdentityService').url;
                 }
-
-                return this.services.find(s => s.name === 'IdentityService').url;
             })
         );
     }
 
+    private baseDiscoveryServiceUrl : string;
+    get discoveryServiceEndpoint(): string {
+        if(this.baseDiscoveryServiceUrl === undefined) {
+            const identityRootUrl: string = this.configService.getIdentityServiceRoot();
```

Comment:
In the oidc client, there is a metadataService object that we might consider leveraging: https://github.com/IdentityModel/oidc-client-js/blob/dev/src/MetadataService.js.

Reply:
ill check this out

Reply:
I see this js object will parse the metadata of the document into a consumable object.  Yeah i can see fitting this in